corehq/apps/app_manager/app_translations/parser.py

```diff
@@ -0,0 +1,112 @@
+        # expected context format
+        # index:Module or Form: sheet name for module/form: unique id
+        context_regex = r'^(\d+):(Module|Form):(\w+):(\w+)$'
+        ws.append(["Type", "sheet_name", self.key_lang_str, self.source_lang_str, 'unique_id'])
+        for po_entry in po_entries:
+            context = po_entry.msgctxt
+            _index, _type, _sheet_name, _unique_id = re.match(context_regex, context).groups()
+            ws.append([_type, _sheet_name, po_entry.msgid, po_entry.msgstr, _unique_id])
+            self.translations[ws.title].append(
+                {
+                    'Type': _type,
+                    'sheet_name': _sheet_name,
+                    self.key_lang_str: po_entry.msgid,
+                    self.source_lang_str: po_entry.msgstr,
+                    'unique_id': _unique_id
+                }
+            )
+
+    def _parse_module_sheet(self, ws, po_entries):
+        # expected context format
+        # index: case property: list/detail
+        context_regex = r'^(\d+):(.+):(list|detail)$'
+        ws.append(["case_property", "list_or_detail", self.key_lang_str, self.source_lang_str])
+        for po_entry in po_entries:
+            context = po_entry.msgctxt
+            _index, _case_property, _list_or_detail = re.match(context_regex, context).groups()
+            ws.append([_case_property, _list_or_detail, po_entry.msgid, po_entry.msgstr])
+            self.translations[ws.title].append(
+                {
+                    'case_property': _case_property,
+                    'list_or_detail': _list_or_detail,
+                    self.key_lang_str: po_entry.msgid,
+                    self.source_lang_str: po_entry.msgstr
+                }
+            )
+
+    def _parse_form_sheet(self, ws, po_entries):
+        # expected context regex
+        # index: label
+        context_regex = r'^(\d+):(.+)$'
+        ws.append(["label", self.key_lang_str, self.source_lang_str])
+        for po_entry in po_entries:
+            context = po_entry.msgctxt
+            _index, _label = re.match(context_regex, context).groups()
+            ws.append([_label, po_entry.msgid, po_entry.msgstr])
+            self.translations[ws.title].append(
+                {
+                    'label': _label,
+                    self.key_lang_str: po_entry.msgid,
+                    self.source_lang_str: po_entry.msgstr
+                }
+            )
+
+    def result_file_name(self, version):
+        return ("TransifexTranslations {}-{}:v{} {}.xlsx".format(
+            self.transifex.key_lang, self.transifex.source_lang,
+            version, datetime.utcnow().strftime(DATETIME_FORMAT))
+        )
+
+    def _generate_sheets(self, wb, resource_slugs):
+        version = self.transifex.version
+        all_translations = self.transifex.get_translations(version, resource_slugs)
+        for resource_name in all_translations:
+            po_entries = all_translations[resource_name]
```

Comment:
You could also do:
```python
for resource_name, po_entries in all_translations.items():
```
or just make `all_translations` a list of tuples if you don't need the dictionary functionality anyways.

Reply:
I remember using it and it was causing an issue. let me try again.

list of tuples? How would that be?
need the dict where resource_name is mapped to a list of translations/PoEntries

Reply:
updated to use the for loop with items.

Reply:
> list of tuples? How would that be?

I mean that these two are equivalent:

```python
po_entries = {}
for resource_slug in resource_slugs:
    po_entries[resource_slug] = self.client.get_translation(resource_slug, self.source_lang, self.lock_translations)
return po_entries

for resource_name, po_entries in get_translations().items():
```

```python
po_entries = [] 
for resource_slug in resource_slugs:
    po_entries.append((resource_slug, self.client.get_translation(resource_slug, self.source_lang, self.lock_translations)))
return po_entries

for resource_name, po_entries in get_translations():
```
Using a dictionary lets you jump to specific entries in the set, or more easily modify them after the fact, but in this case, it iterates over the whole set using `items()`, which converts the dictionary into a list of tuples, so you could also just build a list of tuples directly.  They're really pretty much the same though - I don't think it matters.